# Tech Challenge – Fase 3 | Data Understanding

## Objetivo
Nesse notebook iremos realizar a análise inicial das três pesquisas mais recentes do State of Data Brasil, avaliando estrutura, dimensões, tipos de dados, valores ausentes, duplicidades e diferenças de schema entre os anos.

Esta etapa servirá como base para a definição do pipeline de dados e para a posterior harmonização das pesquisas nas camadas Bronze, Silver e Gold na AWS.

In [2]:
import pandas as pd
import numpy as np

In [3]:
df_2023 = pd.read_csv('/content/State_of_data_BR_2023_Kaggle - df_survey_2023.csv')

df_2024 = pd.read_csv('/content/Final Dataset - State of Data 2024 - Kaggle - df_survey_2024.csv')

df_2025 = pd.read_csv('/content/Final Dataset - State of Data 2025-2026 - Kaggle.csv')

## 1. Visão Geral das Bases

Nesta etapa, são avaliadas as dimensões e a estrutura inicial das três pesquisas utilizadas no projeto, permitindo identificar diferenças entre os anos antes da etapa de harmonização dos dados.

In [4]:
print("Dimensões das bases:\n")

print(f"2023-2024: {df_2023.shape[0]} linhas × {df_2023.shape[1]} colunas")
print(f"2024-2025: {df_2024.shape[0]} linhas × {df_2024.shape[1]} colunas")
print(f"2025-2026: {df_2025.shape[0]} linhas × {df_2025.shape[1]} colunas")

Dimensões das bases:

2023-2024: 5293 linhas × 399 colunas
2024-2025: 5217 linhas × 403 colunas
2025-2026: 3495 linhas × 388 colunas


### 1.1 Amostra dos Dados

A visualização das primeiras linhas permite verificar o formato dos registros, os padrões de nomenclatura das variáveis e a forma como as respostas foram disponibilizadas em cada edição da pesquisa.

In [5]:
df_2023.head()

,"('P0', 'id')","('P1_a ', 'Idade')","('P1_a_1 ', 'Faixa idade')","('P1_b ', 'Genero')","('P1_c ', 'Cor/raca/etnia')","('P1_d ', 'PCD')","('P1_e ', 'experiencia_profissional_prejudicada')","('P1_e_1 ', 'Não acredito que minha experiência profissional seja afetada')","('P1_e_2 ', 'Experiencia prejudicada devido a minha Cor Raça Etnia')","('P1_e_3 ', 'Experiencia prejudicada devido a minha identidade de gênero')",...,"('P8_d_3 ', 'Entrando em contato com os times de negócio para definição do problema, identificar a solução e apresentação de resultados.')","('P8_d_4 ', 'Desenvolvendo modelos de Machine Learning com o objetivo de colocar em produção em sistemas (produtos de dados).')","('P8_d_5 ', 'Colocando modelos em produção, criando os pipelines de dados, APIs de consumo e monitoramento.')","('P8_d_6 ', 'Cuidando da manutenção de modelos de Machine Learning já em produção, atuando no monitoramento, ajustes e refatoração quando necessário.')","('P8_d_7 ', 'Realizando construções de dashboards em ferramentas de BI como PowerBI, Tableau, Looker, Qlik, etc.')","('P8_d_8 ', 'Utilizando ferramentas avançadas de estatística como SAS, SPSS, Stata etc, para realizar análises.')","('P8_d_9 ', 'Criando e dando manutenção em ETLs, DAGs e automações de pipelines de dados.')","('P8_d_10 ', 'Criando e gerenciando soluções de Feature Store e cultura de MLOps.')","('P8_d_11 ', 'Criando e mantendo a infra que meus modelos e soluções rodam (clusters, servidores, API, containers, etc.)')","('P8_d_12 ', 'Treinando e aplicando LLM's para solucionar problemas de negócio.')"
0,001b2d1qtli8t9z7oqgdhj001b2d4i0g,31,30-34,Masculino,Branca,Não,NaN,NaN,NaN,NaN,...,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
1,0026aa3fwd78u0026asg7456tfkjg2cs,30,30-34,Masculino,Branca,Não,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,00r21rb9pusd1b0v7ew00r21rw3dy69w,37,35-39,Feminino,Amarela,Não,"Sim, acredito que a minha a experiência profis...",0.0,0.0,1.0,...,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
3,00urm3jf2cek12w6ygue00urm3jzd17j,22,22-24,Masculino,Preta,Não,Não acredito que minha experiência profissiona...,1.0,0.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,00v0az4g792svil00vn6y1kfm9hq8vy9,34,30-34,Masculino,Branca,Não,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [6]:
df_2024.head()

,0.a_token,0.d_data/hora_envio,1.a_idade,1.a.1_faixa_idade,1.b_genero,1.c_cor/raca/etnia,1.d_pcd,1.e_experiencia_profissional_prejudicada,1.e.1_Não acredito que minha experiência profissional seja afetada,"1.e.2_Sim, devido a minha Cor/Raça/Etnia",...,"8.d.3_Entrando em contato com os times de negócio para definição do problema, identificar a solução e apresentação de resultados.",8.d.4_Desenvolvendo modelos de Machine Learning com o objetivo de colocar em produção em sistemas (produtos de dados).,"8.d.5_Colocando modelos em produção, criando os pipelines de dados, APIs de consumo e monitoramento.","8.d.6_Cuidando da manutenção de modelos de Machine Learning já em produção, atuando no monitoramento, ajustes e refatoração quando necessário.","8.d.7_Realizando construções de dashboards em ferramentas de BI como PowerBI, Tableau, Looker, Qlik, etc.","8.d.8_Utilizando ferramentas avançadas de estatística como SAS, SPSS, Stata etc, para realizar análises.","8.d.9_Criando e dando manutenção em ETLs, DAGs e automações de pipelines de dados.",8.d.10_Criando e gerenciando soluções de Feature Store e cultura de MLOps.,"8.d.11_Criando e mantendo a infra que meus modelos e soluções rodam (clusters, servidores, API, containers, etc.)",8.d.12_Treinando e aplicando LLM's para solucionar problemas de negócio.
0,reb94rv0msth7q4nreb94riaq80iz3yi,16/10/2024 11:19:17,18,17-21,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,1zc66g69jjt49y32l1zc66g8wqj79m4e,16/10/2024 20:45:31,18,17-21,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,uu99wmam4n5kc2uu99wmydf0rk7l58f7,17/10/2024 18:10:59,18,17-21,Masculino,Parda,Não,Não acredito que minha experiência profissiona...,1.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,3ynsw7z0hl5hhpbfvaeqk73ynsw7z04l,22/10/2024 18:03:00,18,17-21,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,v6bji2ct5xckpl1uouv6bjiotkrf3b4f,23/10/2024 01:33:23,18,17-21,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [7]:
df_2025.head()

,0.a_token,0.d_data/hora_envio,1.a_idade,1.a.1_faixa_idade,1.b_genero,1.c_cor/raca/etnia,1.d_pcd,1.e_experiencia_profissional_prejudicada,1.e.1_Não acredito que minha experiência profissional seja afetada,"1.e.2_Sim, devido a minha Cor/Raça/Etnia",...,8.d.4_Desenvolvendo modelos de Machine Learning com o objetivo de colocar em produção em sistemas (produtos de dados).,"8.d.5_Colocando modelos em produção, criando os pipelines de dados, APIs de consumo e monitoramento.","8.d.6_Cuidando da manutenção de modelos de Machine Learning já em produção, atuando no monitoramento, ajustes e refatoração quando necessário.","8.d.7_Realizando construções de dashboards em ferramentas de BI como PowerBI, Tableau, Looker, Qlik, etc.","8.d.8_Utilizando ferramentas avançadas de estatística como SAS, SPSS, Stata etc, para realizar análises.","8.d.9_Criando e dando manutenção em ETLs, DAGs e automações de pipelines de dados.",8.d.10_Criando e gerenciando soluções de Feature Store e cultura de MLOps.,"8.d.11_Criando e mantendo a infra que meus modelos e soluções rodam (clusters, servidores, API, containers, etc.)",8.d.12_Treinando e aplicando LLM's para solucionar problemas de negócio.,8.d_maior_tempo_gasto_como_ds
0,apeczcwvds75ba5hml4apec2pb35hflw,20/10/2025 10:20:11,35,35-39,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,a27c74vkj7n8vmib88ca27c74sepwcpa,20/10/2025 10:27:09,20,17-21,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,7xpkv4qj7mzvsxemhpjy7xpkv4qu61oi,20/10/2025 10:28:59,35,35-39,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,ugzyokw0p1or7axwzugzyz4ombwfgzoo,20/10/2025 10:30:24,29,25-29,Masculino,Parda,Não,Não acredito que minha experiência profissiona...,1.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,n7aiatzlzonity0sn7aiat75y36iuqni,20/10/2025 10:33:11,18,17-21,Masculino,Preta,Não,Não acredito que minha experiência profissiona...,1.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


### 1.2 Resumo Estrutural

Nesta etapa, são comparadas as dimensões das bases, a quantidade de registros duplicados e a existência de nomes de colunas duplicados. O objetivo é identificar possíveis problemas estruturais antes de qualquer transformação.

In [8]:
resumo_estrutural = pd.DataFrame({
    "Base": ["2023-2024", "2024-2025", "2025-2026"],
    "Linhas": [
        df_2023.shape[0],
        df_2024.shape[0],
        df_2025.shape[0]
    ],
    "Colunas": [
        df_2023.shape[1],
        df_2024.shape[1],
        df_2025.shape[1]
    ],
    "Linhas duplicadas": [
        df_2023.duplicated().sum(),
        df_2024.duplicated().sum(),
        df_2025.duplicated().sum()
    ],
    "Nomes de colunas duplicados": [
        df_2023.columns.duplicated().sum(),
        df_2024.columns.duplicated().sum(),
        df_2025.columns.duplicated().sum()
    ]
})

resumo_estrutural

,Base,Linhas,Colunas,Linhas duplicadas,Nomes de colunas duplicados
0,2023-2024,5293,399,0,0
1,2024-2025,5217,403,2,0
2,2025-2026,3495,388,1,0


**Observação:** As três bases apresentam estruturas distintas em quantidade de registros e variáveis. Foram identificadas 2 linhas duplicadas na pesquisa 2024-2025 e 1 linha duplicada na pesquisa 2025-2026, enquanto a base 2023-2024 não apresentou duplicidades completas. Não foram identificados nomes de colunas duplicados em nenhuma das bases.

As duplicidades serão apenas documentadas nesta etapa exploratória e tratadas posteriormente durante o processo de transformação dos dados.

### 1.3 Tipos de Dados

A distribuição dos tipos de dados é analisada para compreender como as variáveis foram interpretadas na leitura dos arquivos e identificar possíveis necessidades de conversão e padronização nas etapas posteriores do pipeline.

In [9]:
tipos_dados = pd.DataFrame({
    "2023-2024": df_2023.dtypes.value_counts(),
    "2024-2025": df_2024.dtypes.value_counts(),
    "2025-2026": df_2025.dtypes.value_counts()
}).fillna(0).astype(int)

tipos_dados

,2023-2024,2024-2025,2025-2026
bool,0,1,0
int64,2,1,2
float64,328,323,315
object,69,78,71


## 2. Comparação dos Schemas entre as Pesquisas

Como as pesquisas foram realizadas em edições diferentes, é necessário avaliar a consistência dos schemas antes da integração dos dados.

Nesta etapa, são comparados os nomes das variáveis entre as três bases, buscando identificar colunas comuns e diferenças estruturais que deverão ser consideradas posteriormente na harmonização dos dados.

In [10]:
colunas_2023 = set(df_2023.columns)
colunas_2024 = set(df_2024.columns)
colunas_2025 = set(df_2025.columns)

print(f"Colunas em 2023-2024: {len(colunas_2023)}")
print(f"Colunas em 2024-2025: {len(colunas_2024)}")
print(f"Colunas em 2025-2026: {len(colunas_2025)}")

Colunas em 2023-2024: 399
Colunas em 2024-2025: 403
Colunas em 2025-2026: 388


In [11]:
comuns_tres_bases = colunas_2023 & colunas_2024 & colunas_2025

comuns_2023_2024 = colunas_2023 & colunas_2024
comuns_2024_2025 = colunas_2024 & colunas_2025
comuns_2023_2025 = colunas_2023 & colunas_2025

print(f"Colunas comuns às três bases: {len(comuns_tres_bases)}")
print(f"Colunas comuns entre 2023-2024 e 2024-2025: {len(comuns_2023_2024)}")
print(f"Colunas comuns entre 2024-2025 e 2025-2026: {len(comuns_2024_2025)}")
print(f"Colunas comuns entre 2023-2024 e 2025-2026: {len(comuns_2023_2025)}")

Colunas comuns às três bases: 0
Colunas comuns entre 2023-2024 e 2024-2025: 0
Colunas comuns entre 2024-2025 e 2025-2026: 284
Colunas comuns entre 2023-2024 e 2025-2026: 0


### Interpretação da Comparação dos Schemas

A comparação dos nomes das colunas evidencia uma diferença estrutural relevante entre as edições das pesquisas.

As bases de 2024-2025 e 2025-2026 apresentam **284 colunas com nomenclatura idêntica**, indicando maior continuidade no padrão adotado entre essas duas edições.

Por outro lado, a base de 2023-2024 não apresenta nomes de colunas exatamente iguais aos encontrados nas duas pesquisas posteriores. Isso não significa necessariamente que as pesquisas não possuam variáveis equivalentes, mas indica uma **mudança no padrão de nomenclatura das colunas**.

Dessa forma, a integração das três pesquisas não poderá ser realizada apenas pela correspondência direta dos nomes das variáveis. Será necessária uma etapa posterior de **harmonização do schema**, identificando variáveis semanticamente equivalentes entre os diferentes anos antes da consolidação dos dados.

### 2.1 Diferenças entre os Schemas de 2024-2025 e 2025-2026

Como as duas edições mais recentes apresentam maior padronização na nomenclatura das variáveis, são identificadas as colunas exclusivas de cada pesquisa para avaliar alterações realizadas no questionário entre os períodos.

In [12]:
exclusivas_2024 = sorted(colunas_2024 - colunas_2025)
exclusivas_2025 = sorted(colunas_2025 - colunas_2024)

print(f"Colunas exclusivas de 2024-2025: {len(exclusivas_2024)}")
print(f"Colunas exclusivas de 2025-2026: {len(exclusivas_2025)}")

Colunas exclusivas de 2024-2025: 119
Colunas exclusivas de 2025-2026: 104


### 2.2 Inspeção das Variáveis Exclusivas

Embora as pesquisas de 2024-2025 e 2025-2026 compartilhem 284 variáveis com a mesma nomenclatura, também existem variáveis exclusivas em cada edição.

A inspeção dessas colunas permite verificar se as diferenças representam novas perguntas, perguntas removidas ou alterações na nomenclatura e estrutura do questionário.

In [13]:
print("=== Colunas exclusivas de 2024-2025 ===\n")

for coluna in exclusivas_2024:
    print(coluna)

print("\n" + "=" * 80 + "\n")

print("=== Colunas exclusivas de 2025-2026 ===\n")

for coluna in exclusivas_2025:
    print(coluna)

=== Colunas exclusivas de 2024-2025 ===

2.q_empresa_passou_por_layoff_em_2024
2.r_modelo_de_trabalho_atual
2.s_modelo_de_trabalho_ideal
2.t_atitude_em_caso_de_retorno_presencial
3.g.1 Falta de compreensão dos casos de uso
3.g.2 Falta de confiabilidade das saídas (alucinação dos modelos)
3.g.3 Incerteza em relação a regulamentação
3.g.4 Preocupações com segurança e privacidade de dados
3.g.5 Retorno sobre investimento (ROI) não comprovado de IA Generativa
3.g.6 Dados da empresa não estão prontos para uso de IA Generativa
3.g.7 Falta de expertise ou falta de recursos
3.g.8 Alta direção da empresa não vê valor ou não vê como prioridade
3.g.9 Preocupações com propriedade intelectual
3.g_motivos_para_não_usar_ai_generativa_e_llm
4.c.1_Dados relacionais (estruturados em bancos SQL)
4.c.2_Dados armazenados em bancos NoSQL
4.c.3_Imagens
4.c.4_Textos/Documentos
4.c.5_Vídeos
4.c.6_Áudios
4.c.7_Planilhas
4.c.8_Dados georeferenciados
4.c_fonte_de_dado_mais_usada
4.d.10_Scala
4.d.11_Matlab
4.d.12_

### Interpretação das Variáveis Exclusivas

A inspeção inicial demonstra que as diferenças entre as edições de 2024-2025 e 2025-2026 não se limitam à quantidade de colunas. Existem alterações no conjunto de variáveis e nos temas investigados pelas pesquisas.

Entre as colunas exclusivas, observam-se grupos relacionados a tecnologias, ferramentas, infraestrutura de dados e Inteligência Artificial Generativa, indicando evolução do questionário para acompanhar mudanças no mercado de Dados.

Entretanto, a classificação de uma coluna como exclusiva considera apenas a correspondência exata de seu nome. Portanto, parte dessas diferenças também pode estar associada a alterações de nomenclatura ou reformulação de perguntas entre as edições.

Por esse motivo, a harmonização das pesquisas deverá considerar não apenas os nomes das colunas, mas também o significado das variáveis.

### 2.3 Distribuição das Variáveis Exclusivas por Bloco do Questionário

Para facilitar a identificação das áreas do questionário que sofreram alterações entre as edições, as variáveis exclusivas são agrupadas de acordo com o primeiro identificador presente no nome de cada coluna.

In [14]:
from collections import Counter

def obter_bloco(coluna):
    return str(coluna).split(".")[0]

blocos_2024 = Counter(obter_bloco(coluna) for coluna in exclusivas_2024)
blocos_2025 = Counter(obter_bloco(coluna) for coluna in exclusivas_2025)

comparacao_blocos = pd.DataFrame({
    "Exclusivas 2024-2025": pd.Series(blocos_2024),
    "Exclusivas 2025-2026": pd.Series(blocos_2025)
}).fillna(0).astype(int)

comparacao_blocos

,Exclusivas 2024-2025,Exclusivas 2025-2026
2,4,4
3,10,11
4,105,89


### Interpretação da Distribuição por Blocos

As diferenças de schema entre as pesquisas de 2024-2025 e 2025-2026 estão fortemente concentradas no **bloco 4 do questionário**, que reúne 105 das 119 variáveis exclusivas de 2024-2025 e 89 das 104 variáveis exclusivas de 2025-2026.

Os blocos 2 e 3 apresentam quantidade significativamente menor de variáveis exclusivas, indicando maior estabilidade estrutural entre as duas edições nessas seções.

A concentração das alterações no bloco 4 sugere que grande parte da evolução do questionário ocorreu nas perguntas relacionadas ao ecossistema tecnológico e às práticas profissionais, incluindo ferramentas, infraestrutura e temas associados à Inteligência Artificial.

Esse comportamento deverá ser considerado na harmonização dos dados, especialmente nas análises de evolução tecnológica entre os diferentes períodos.

### 2.4 Análise da Estrutura de Nomenclatura da Pesquisa 2023-2024

A pesquisa de 2023-2024 utiliza uma convenção de nomenclatura diferente das edições posteriores. Por esse motivo, a ausência de correspondência exata entre os nomes das colunas não implica necessariamente ausência de variáveis equivalentes.

Nesta etapa, são inspecionados os nomes das primeiras variáveis de cada pesquisa para identificar padrões de nomenclatura e avaliar a necessidade de mapeamento semântico entre as edições.

In [15]:
comparacao_nomes = pd.DataFrame({
    "2023-2024": list(df_2023.columns[:30]),
    "2024-2025": list(df_2024.columns[:30]),
    "2025-2026": list(df_2025.columns[:30])
})

pd.set_option("display.max_colwidth", None)

comparacao_nomes

,2023-2024,2024-2025,2025-2026
0,"('P0', 'id')",0.a_token,0.a_token
1,"('P1_a ', 'Idade')",0.d_data/hora_envio,0.d_data/hora_envio
2,"('P1_a_1 ', 'Faixa idade')",1.a_idade,1.a_idade
3,"('P1_b ', 'Genero')",1.a.1_faixa_idade,1.a.1_faixa_idade
4,"('P1_c ', 'Cor/raca/etnia')",1.b_genero,1.b_genero
5,"('P1_d ', 'PCD')",1.c_cor/raca/etnia,1.c_cor/raca/etnia
6,"('P1_e ', 'experiencia_profissional_prejudicada')",1.d_pcd,1.d_pcd
7,"('P1_e_1 ', 'Não acredito que minha experiência profissional seja afetada')",1.e_experiencia_profissional_prejudicada,1.e_experiencia_profissional_prejudicada
8,"('P1_e_2 ', 'Experiencia prejudicada devido a minha Cor Raça Etnia')",1.e.1_Não acredito que minha experiência profissional seja afetada,1.e.1_Não acredito que minha experiência profissional seja afetada
9,"('P1_e_3 ', 'Experiencia prejudicada devido a minha identidade de gênero')","1.e.2_Sim, devido a minha Cor/Raça/Etnia","1.e.2_Sim, devido a minha Cor/Raça/Etnia"


### Interpretação da Nomenclatura da Pesquisa 2023-2024

A inspeção das primeiras variáveis confirma que a pesquisa de 2023-2024 utiliza uma convenção de nomenclatura diferente das edições posteriores.

Enquanto a edição de 2023-2024 apresenta nomes compostos pelo identificador da pergunta e sua descrição, como `('P1_b ', 'Genero')` e `('P1_c ', 'Cor/raca/etnia')`, as pesquisas seguintes utilizam identificadores padronizados, como `1.b_genero` e `1.c_cor/raca/etnia`.

Também foi identificado que a posição das variáveis não permanece necessariamente igual entre as edições, devido à inclusão, remoção ou reorganização de perguntas. Portanto, o alinhamento das colunas não poderá ser realizado apenas pela posição ou pela igualdade textual dos nomes.

A futura etapa de transformação deverá realizar um **mapeamento semântico das variáveis**, permitindo associar perguntas equivalentes entre os diferentes anos e construir um schema padronizado para a camada Silver.

## 3. Análise de Valores Ausentes

As pesquisas possuem grande quantidade de variáveis e apresentam perguntas condicionais, nas quais determinadas questões são respondidas apenas por grupos específicos de participantes.

Por esse motivo, a presença de valores ausentes deve ser analisada antes de qualquer estratégia de tratamento, evitando interpretar automaticamente valores nulos como problemas de qualidade dos dados.

In [16]:
resumo_nulos = pd.DataFrame({
    "Base": ["2023-2024", "2024-2025", "2025-2026"],

    "Total de células": [
        df_2023.size,
        df_2024.size,
        df_2025.size
    ],

    "Células nulas": [
        df_2023.isna().sum().sum(),
        df_2024.isna().sum().sum(),
        df_2025.isna().sum().sum()
    ]
})

resumo_nulos["Percentual de nulos (%)"] = (
    resumo_nulos["Células nulas"] /
    resumo_nulos["Total de células"] * 100
).round(2)

resumo_nulos

,Base,Total de células,Células nulas,Percentual de nulos (%)
0,2023-2024,2111907,1197052,56.68
1,2024-2025,2102451,1202637,57.20
2,2025-2026,1356060,839991,61.94


### Interpretação dos Valores Ausentes

As três pesquisas apresentam elevada proporção de valores ausentes, correspondendo a **56,68%** das células em 2023-2024, **57,20%** em 2024-2025 e **61,04%** em 2025-2026.

Entretanto, esses percentuais não devem ser interpretados isoladamente como problemas de qualidade dos dados. O questionário contém perguntas condicionais e grupos de variáveis aplicáveis apenas a determinados perfis ou respostas anteriores, fazendo com que a ausência de informação possa representar um comportamento esperado da pesquisa.

Observa-se também aumento do percentual global de valores ausentes na edição mais recente. Antes da definição de qualquer estratégia de tratamento, será necessário avaliar os valores nulos no contexto de cada variável e de sua função no questionário.

Dessa forma, não será aplicada nesta etapa uma estratégia generalizada de preenchimento ou remoção de valores ausentes. O tratamento será definido posteriormente de acordo com as variáveis selecionadas para as análises e com as regras de negócio associadas a cada indicador.

### 3.1 Variáveis com Maior Percentual de Valores Ausentes

Para compreender melhor a distribuição dos valores ausentes, são identificadas as variáveis com maior percentual de dados nulos em cada edição da pesquisa.

In [17]:
def top_colunas_nulas(df, n=10):
    percentual = (df.isna().mean() * 100).round(2)

    return (
        percentual
        .sort_values(ascending=False)
        .head(n)
        .reset_index()
        .rename(columns={
            "index": "Variável",
            0: "Percentual de nulos (%)"
        })
    )

nulos_2023 = top_colunas_nulas(df_2023)
nulos_2024 = top_colunas_nulas(df_2024)
nulos_2025 = top_colunas_nulas(df_2025)

In [18]:
print("2023-2024")
display(nulos_2023)

print("\n2024-2025")
display(nulos_2024)

print("\n2025-2026")
display(nulos_2025)

2023-2024


,Variável,Percentual de nulos (%)
0,"('P5_d ', 'Como tem sido a busca por um emprego na área de dados?')",94.60
1,"('P5_c ', 'Há quanto tempo você busca uma oportunidade na área de dados?')",94.58
2,"('P5_b ', 'Qual oportunidade você está buscando?')",94.58
3,"('P5_a ', 'Qual seu objetivo na área de dados?')",90.08
4,"('P6_g ', 'Quais as ferramentas de gestão de Qualidade de dados, Metadados e catálogo de dados você utiliza no trabalho?')",89.51
5,"('P6_f ', 'Qual tecnologia utilizada como plataforma do Data Warehouse?')",86.04
6,"('P6_d ', 'Qual tecnologia utilizada como plataforma do Data Lake?')",85.98
7,"('P8_d_11 ', 'Criando e mantendo a infra que meus modelos e soluções rodam (clusters, servidores, API, containers, etc.)')",85.87
8,"('P8_d_12 ', 'Treinando e aplicando LLM's para solucionar problemas de negócio.')",85.87
9,"('P8_3 ', 'Quais dessas tecnologias fazem parte do seu dia a dia como cientista de dados?')",85.87



2024-2025


,Variável,Percentual de nulos (%)
0,1.h_pais_onde_mora,97.34
1,5.b_oportunidade_buscada,94.52
2,5.d_experiencia_em_processos_seletivos,94.46
3,5.c_tempo_em_busca_de_oportunidade,94.44
4,5.a_objetivo_na_area_de_dados,89.63
5,6.g_ferramentas_de_qualidade_de_dados_(dia_a_dia),89.27
6,6.f_tecnologia_data_warehouse,86.08
7,6.d_tecnologia_data_lake,85.91
8,"8.d.11_Criando e mantendo a infra que meus modelos e soluções rodam (clusters, servidores, API, containers, etc.)",85.18
9,8.c_tecnologias_ds,85.18



2025-2026


,Variável,Percentual de nulos (%)
0,1.h_pais_onde_mora,96.51
1,5.b_oportunidade_buscada,94.76
2,5.d_experiencia_em_processos_seletivos,94.65
3,5.c_tempo_em_busca_de_oportunidade,94.51
4,5.a_objetivo_na_area_de_dados,88.15
5,6.d_tecnologia_data_lake,88.13
6,6.f_tecnologia_data_warehouse,87.98
7,"8.c.3_Ambientes de desenvolvimento local (R-studio, JupyterLab, Anaconda).",85.98
8,"8.c.4_Ambientes de desenvolvimento na nuvem (Google Colab, AWS Sagemaker, Kaggle Notebooks etc).",85.98
9,"8.c.2_Planilhas (Excel, Google Sheets etc).",85.98


### Interpretação das Variáveis com Maior Percentual de Nulos

A análise das variáveis com maior percentual de valores ausentes reforça que a ausência de dados está concentrada, em grande parte, em perguntas específicas ou condicionais do questionário.

Nas edições de 2024-2025 e 2025-2026, algumas das maiores taxas de ausência aparecem em variáveis relacionadas à busca por oportunidades profissionais, experiência em processos seletivos e determinadas tecnologias ou ambientes de trabalho.

Também é possível observar padrões semelhantes de ausência entre as duas edições mais recentes, o que sugere continuidade na estrutura condicional de determinadas perguntas.

Portanto, elevados percentuais de valores nulos não serão tratados automaticamente como problemas de qualidade. A estratégia de tratamento deverá considerar o contexto de cada variável e sua utilização nas análises, evitando remoções ou imputações indiscriminadas.

## 4. Identificação das Variáveis Relevantes para o Desafio

Após a avaliação estrutural das bases, a próxima etapa consiste em identificar os grupos de variáveis necessários para responder às questões de negócio propostas no Tech Challenge.

Como as pesquisas possuem centenas de variáveis e apresentam mudanças de schema entre as edições, a harmonização será orientada pelas necessidades analíticas do projeto, priorizando os atributos relacionados aos principais eixos de análise.

### 4.1 Eixos Analíticos

Os principais eixos inicialmente considerados são:

- **Perfil demográfico:** idade, gênero e raça/etnia;
- **Localização:** estado, região e país;
- **Perfil profissional:** cargo, área de atuação, experiência e senioridade;
- **Remuneração:** faixa salarial e aspectos relacionados;
- **Modelo de trabalho:** presencial, híbrido e remoto;
- **Tecnologias:** linguagens, bancos de dados, ferramentas, cloud e tecnologias utilizadas;
- **Inteligência Artificial:** adoção, utilização e impactos de IA/IA Generativa;
- **Mercado de trabalho:** oportunidades, empregabilidade e características profissionais.

A seleção definitiva das variáveis será realizada considerando a disponibilidade e a equivalência semântica entre as três edições da pesquisa.

In [19]:
termos_interesse = [
    "genero",
    "idade",
    "raca",
    "etnia",
    "estado",
    "regiao",
    "cargo",
    "salario",
    "senior",
    "experiencia",
    "trabalho",
    "remoto",
    "hibrido",
    "tecnologia",
    "linguagem",
    "cloud",
    "inteligencia artificial",
    "generativa",
    "llm"
]

def buscar_variaveis(df, termos):
    resultados = []

    for coluna in df.columns:
        nome = str(coluna).lower()

        if any(termo in nome for termo in termos):
            resultados.append(coluna)

    return resultados

variaveis_2023 = buscar_variaveis(df_2023, termos_interesse)
variaveis_2024 = buscar_variaveis(df_2024, termos_interesse)
variaveis_2025 = buscar_variaveis(df_2025, termos_interesse)

print(f"Variáveis candidatas em 2023-2024: {len(variaveis_2023)}")
print(f"Variáveis candidatas em 2024-2025: {len(variaveis_2024)}")
print(f"Variáveis candidatas em 2025-2026: {len(variaveis_2025)}")

Variáveis candidatas em 2023-2024: 107
Variáveis candidatas em 2024-2025: 99
Variáveis candidatas em 2025-2026: 98


### Interpretação da Busca por Variáveis Relevantes

A busca inicial por palavras-chave identificou **107 variáveis candidatas em 2023-2024, 99 em 2024-2025 e 98 em 2025-2026** relacionadas aos principais eixos analíticos definidos para o projeto.

Essa seleção possui caráter exploratório e não representa ainda o conjunto definitivo de atributos que será utilizado nas análises. Como a identificação foi realizada por correspondência textual nos nomes das colunas, algumas variáveis relevantes podem não ter sido capturadas, enquanto outras podem ter sido incluídas apenas por apresentarem termos semelhantes.

A definição do schema analítico será realizada posteriormente por meio da avaliação semântica das variáveis e de sua aderência às questões de negócio do Tech Challenge.

### 4.2 Inspeção das Variáveis Candidatas

As variáveis identificadas pela busca textual são inspecionadas para avaliar sua aderência aos eixos analíticos do projeto e auxiliar na posterior definição do conjunto de atributos que será harmonizado entre as pesquisas.

In [20]:
print("=== 2023-2024 ===")
for coluna in variaveis_2023:
    print(coluna)

print("\n=== 2024-2025 ===")
for coluna in variaveis_2024:
    print(coluna)

print("\n=== 2025-2026 ===")
for coluna in variaveis_2025:
    print(coluna)

=== 2023-2024 ===
('P1_a ', 'Idade')
('P1_a_1 ', 'Faixa idade')
('P1_b ', 'Genero')
('P1_c ', 'Cor/raca/etnia')
('P1_e ', 'experiencia_profissional_prejudicada')
('P1_e_2 ', 'Experiencia prejudicada devido a minha Cor Raça Etnia')
('P1_e_3 ', 'Experiencia prejudicada devido a minha identidade de gênero')
('P1_e_4 ', 'Experiencia prejudicada devido ao fato de ser PCD')
('P1_f_1', 'Quantidade de oportunidades de emprego/vagas recebidas')
('P1_f_2', 'Senioridade das vagas recebidas em relação à sua experiência')
('P1_f_4', 'Oportunidades de progressão de carreira')
('P1_f_5', 'Velocidade de progressão de carreira')
('P1_f_6', 'Nível de cobrança no trabalho/Stress no trabalho')
('P1_f_8', 'Relação com outros membros da empresa, em momentos de trabalho')
('P1_f_9', 'Relação com outros membros da empresa, em momentos de integração e outros momentos fora do trabalho')
('P1_i ', 'Estado onde mora')
('P1_i_2 ', 'Regiao onde mora')
('P1_j ', 'Mudou de Estado?')
('P1_k ', 'Regiao de origem')
('P2

### Interpretação das Variáveis Candidatas

A inspeção das variáveis candidatas confirma a disponibilidade de atributos relacionados aos principais eixos analíticos do projeto, incluindo perfil demográfico, localização, carreira, experiência profissional, modelo de trabalho, tecnologias e adoção de Inteligência Artificial.

Também foram identificadas variáveis semanticamente equivalentes cujos identificadores foram alterados entre as edições. Por exemplo, a variável relacionada ao modelo de trabalho atual aparece como `P2_r` em 2023-2024, `2.r_modelo_de_trabalho_atual` em 2024-2025 e `2.q_modelo_de_trabalho_atual` em 2025-2026.

Esse comportamento reforça que a harmonização deverá ser orientada pelo significado das variáveis, e não apenas pelos nomes originais das colunas.

A busca por palavras-chave foi utilizada como mecanismo exploratório para localizar atributos potencialmente relevantes. A seleção definitiva será orientada pelas questões de negócio do Tech Challenge.

## 5. Mapeamento das Questões de Negócio

A seleção das variáveis que comporão as camadas analíticas será orientada pelas questões de negócio propostas no Tech Challenge.

Essa abordagem evita a harmonização indiscriminada de todas as centenas de variáveis disponíveis e permite concentrar o processamento nos atributos necessários para gerar análises relevantes sobre o mercado brasileiro de Dados.

In [21]:
questoes_negocio = pd.DataFrame({
    "Questão do Tech Challenge": [
        "Como está estruturado o mercado brasileiro de Dados?",
        "Quais perfis profissionais são mais valorizados pelo mercado?",
        "Qual é o cenário de diversidade de gênero nas carreiras de dados?",
        "Quais tecnologias apresentam maior adoção entre os profissionais?",
        "Qual é o índice de adoção de Inteligência Artificial e seu impacto?",
        "Existem diferenças relevantes entre regiões, senioridades ou modelos de trabalho?",
        "Quais oportunidades e desafios podem ser identificados para empresas que desejam investir em Dados e Inteligência Artificial?"
    ],

    "Eixo Analítico": [
        "Estrutura do mercado",
        "Perfil profissional",
        "Diversidade",
        "Tecnologias",
        "Inteligência Artificial",
        "Segmentação",
        "Oportunidades e desafios"
    ],

    "Variáveis necessárias": [
        "Cargo, experiência, senioridade, formação, remuneração e situação profissional",
        "Cargo, senioridade, experiência, remuneração e oportunidades profissionais",
        "Gênero, cargo, senioridade, remuneração e experiência profissional",
        "Linguagens, cloud, bancos de dados, BI e demais tecnologias",
        "IA Generativa, LLMs, ChatGPT/Copilot, utilização e adoção pelas empresas",
        "Região, estado, senioridade, modelo de trabalho e indicadores profissionais",
        "Empregabilidade, tecnologias, IA, progressão de carreira, qualificação e características do mercado"
    ]
})

questoes_negocio

questoes_negocio

,Questão do Tech Challenge,Eixo Analítico,Variáveis necessárias
0,Como está estruturado o mercado brasileiro de Dados?,Estrutura do mercado,"Cargo, experiência, senioridade, formação, remuneração e situação profissional"
1,Quais perfis profissionais são mais valorizados pelo mercado?,Perfil profissional,"Cargo, senioridade, experiência, remuneração e oportunidades profissionais"
2,Qual é o cenário de diversidade de gênero nas carreiras de dados?,Diversidade,"Gênero, cargo, senioridade, remuneração e experiência profissional"
3,Quais tecnologias apresentam maior adoção entre os profissionais?,Tecnologias,"Linguagens, cloud, bancos de dados, BI e demais tecnologias"
4,Qual é o índice de adoção de Inteligência Artificial e seu impacto?,Inteligência Artificial,"IA Generativa, LLMs, ChatGPT/Copilot, utilização e adoção pelas empresas"
5,"Existem diferenças relevantes entre regiões, senioridades ou modelos de trabalho?",Segmentação,"Região, estado, senioridade, modelo de trabalho e indicadores profissionais"
6,Quais oportunidades e desafios podem ser identificados para empresas que desejam investir em Dados e Inteligência Artificial?,Oportunidades e desafios,"Empregabilidade, tecnologias, IA, progressão de carreira, qualificação e características do mercado"


### Interpretação do Mapeamento das Questões de Negócio

O mapeamento foi estruturado a partir das questões propostas no enunciado oficial do Tech Challenge, relacionando cada pergunta de negócio aos principais eixos analíticos e aos grupos de variáveis necessários para respondê-la.

Essa estrutura será utilizada como referência para a seleção e harmonização das variáveis das três edições da pesquisa, evitando o processamento indiscriminado de atributos que não contribuam diretamente para os objetivos analíticos do projeto.

A partir desse mapeamento, o pipeline será estruturado para transformar os dados brutos em conjuntos analíticos capazes de suportar indicadores, visualizações, insights e recomendações estratégicas.

### 5.1 Identificação das Variáveis de Remuneração

A remuneração constitui uma dimensão relevante para a compreensão do mercado brasileiro de Dados e para a análise das diferenças entre perfis profissionais, senioridades, regiões e modelos de trabalho.

Nesta etapa, são identificadas as variáveis relacionadas à remuneração nas três edições da pesquisa, buscando posteriormente estabelecer correspondências semânticas entre os diferentes schemas.

In [22]:
def localizar_colunas(df, termos):
    return [
        coluna for coluna in df.columns
        if any(termo in str(coluna).lower() for termo in termos)
    ]

termos_remuneracao = [
    "salario",
    "salarial",
    "remuneracao",
    "remuneração",
    "renda"
]

remuneracao_2023 = localizar_colunas(df_2023, termos_remuneracao)
remuneracao_2024 = localizar_colunas(df_2024, termos_remuneracao)
remuneracao_2025 = localizar_colunas(df_2025, termos_remuneracao)

print("=== 2023-2024 ===")
for coluna in remuneracao_2023:
    print(coluna)

print("\n=== 2024-2025 ===")
for coluna in remuneracao_2024:
    print(coluna)

print("\n=== 2025-2026 ===")
for coluna in remuneracao_2025:
    print(coluna)

=== 2023-2024 ===
('P2_h ', 'Faixa salarial')
('P2_o_1 ', 'Remuneração/Salário')

=== 2024-2025 ===
2.h_faixa_salarial
2.l.1_Remuneração/Salário
2.o.1_Remuneração/Salário

=== 2025-2026 ===
2.h_faixa_salarial
2.l.1_Remuneração/Salário
2.o.1_Remuneração/Salário


### Interpretação das Variáveis de Remuneração

A análise identificou variáveis relacionadas à remuneração nas três edições da pesquisa.

A variável de **faixa salarial** apresenta correspondência semântica clara entre os períodos, sendo representada por `P2_h` em 2023-2024 e por `2.h_faixa_salarial` nas edições de 2024-2025 e 2025-2026.

Essa correspondência torna a faixa salarial uma candidata adequada para compor o schema harmonizado, permitindo análises relacionadas à remuneração ao longo dos diferentes períodos.

Também foram identificadas outras variáveis associadas a remuneração e salário. Entretanto, sua utilização dependerá da interpretação do contexto das respectivas perguntas, evitando combinar atributos que, apesar de possuírem termos semelhantes, representem conceitos distintos.

### 5.2 Validação da Faixa Salarial entre as Edições

Após identificar uma possível correspondência semântica para a variável de faixa salarial, são analisadas as categorias existentes em cada edição para verificar a compatibilidade dos valores antes de sua futura harmonização.

In [24]:
coluna_salario_2023 = [
    coluna for coluna in df_2023.columns
    if "Faixa salarial" in str(coluna)
][0]

print("Coluna encontrada:")
print(repr(coluna_salario_2023))

Coluna encontrada:
"('P2_h ', 'Faixa salarial')"


In [25]:
print("=== 2023-2024 ===")
print(df_2023[coluna_salario_2023].value_counts(dropna=False))

print("\n=== 2024-2025 ===")
print(df_2024['2.h_faixa_salarial'].value_counts(dropna=False))

print("\n=== 2025-2026 ===")
print(df_2025['2.h_faixa_salarial'].value_counts(dropna=False))

=== 2023-2024 ===
('P2_h ', 'Faixa salarial')
de R$ 8.001/mês a R$ 12.000/mês     1026
de R$ 4.001/mês a R$ 6.000/mês       745
de R$ 12.001/mês a R$ 16.000/mês     650
de R$ 6.001/mês a R$ 8.000/mês       637
NaN                                  540
de R$ 3.001/mês a R$ 4.000/mês       352
de R$ 16.001/mês a R$ 20.000/mês     328
de R$ 2.001/mês a R$ 3.000/mês       288
de R$ 1.001/mês a R$ 2.000/mês       215
de R$ 20.001/mês a R$ 25.000/mês     195
de R$ 25.001/mês a R$ 30.000/mês     128
de R$ 30.001/mês a R$ 40.000/mês      86
Acima de R$ 40.001/mês                72
Menos de R$ 1.000/mês                 30
de R$ 101/mês a R$ 2.000/mês           1
Name: count, dtype: int64

=== 2024-2025 ===
2.h_faixa_salarial
de R$ 8.001/mês a R$ 12.000/mês     1080
de R$ 12.001/mês a R$ 16.000/mês     716
de R$ 6.001/mês a R$ 8.000/mês       656
de R$ 4.001/mês a R$ 6.000/mês       593
de R$ 16.001/mês a R$ 20.000/mês     456
NaN                                  354
de R$ 3.001/mês a R$ 4.000/mê

### Interpretação da Compatibilidade das Faixas Salariais

A comparação das categorias de faixa salarial demonstra elevada consistência entre as três edições da pesquisa. As principais faixas de remuneração permanecem equivalentes ao longo dos períodos, permitindo a utilização dessa dimensão em análises temporais e comparativas.

Foram identificadas, entretanto, ocorrências pontuais com categorias aparentemente inconsistentes: uma ocorrência em 2023-2024 (`de R$ 101/mês a R$ 2.000/mês`) e uma ocorrência em 2025-2026 (`de R$ 25.001/mês a R$ 3000/mês`).

Esses casos serão preservados na camada Bronze e avaliados durante a transformação para a camada Silver, onde serão aplicadas regras explícitas de limpeza e padronização.

Dessa forma, a variável de faixa salarial apresenta condições adequadas para compor o schema harmonizado das três pesquisas sob o nome padronizado `faixa_salarial`.

## 6. Definição Inicial do Schema Harmonizado

A partir da análise estrutural e das questões de negócio, inicia-se a definição do schema que será utilizado para integrar as três edições da pesquisa.

O objetivo é estabelecer nomes padronizados para variáveis semanticamente equivalentes, independentemente da nomenclatura utilizada nos arquivos originais.

Esse mapeamento servirá posteriormente como referência para as transformações realizadas na camada Silver.

In [26]:
schema_harmonizado = pd.DataFrame({
    "Variável Padronizada": [
        "genero",
        "raca_etnia",
        "faixa_salarial",
        "modelo_trabalho"
    ],

    "2023-2024": [
        "P1_b - Genero",
        "P1_c - Cor/raca/etnia",
        "P2_h - Faixa salarial",
        "P2_r - Atualmente qual a sua forma de trabalho?"
    ],

    "2024-2025": [
        "1.b_genero",
        "1.c_cor/raca/etnia",
        "2.h_faixa_salarial",
        "2.r_modelo_de_trabalho_atual"
    ],

    "2025-2026": [
        "1.b_genero",
        "1.c_cor/raca/etnia",
        "2.h_faixa_salarial",
        "2.q_modelo_de_trabalho_atual"
    ]
})

schema_harmonizado

,Variável Padronizada,2023-2024,2024-2025,2025-2026
0,genero,P1_b - Genero,1.b_genero,1.b_genero
1,raca_etnia,P1_c - Cor/raca/etnia,1.c_cor/raca/etnia,1.c_cor/raca/etnia
2,faixa_salarial,P2_h - Faixa salarial,2.h_faixa_salarial,2.h_faixa_salarial
3,modelo_trabalho,P2_r - Atualmente qual a sua forma de trabalho?,2.r_modelo_de_trabalho_atual,2.q_modelo_de_trabalho_atual


### 6.1 Validação das Variáveis Harmonizadas

Antes da utilização das variáveis no schema consolidado, são comparadas as categorias presentes em cada edição da pesquisa.

Essa validação permite confirmar se variáveis semanticamente equivalentes também apresentam conteúdos compatíveis entre os períodos, identificando eventuais diferenças de codificação ou categorias que deverão ser tratadas na camada Silver.

In [27]:
coluna_modelo_2023 = [
    coluna for coluna in df_2023.columns
    if "Atualmente qual a sua forma de trabalho?" in str(coluna)
][0]

print("=== 2023-2024 ===")
print(df_2023[coluna_modelo_2023].value_counts(dropna=False))

print("\n=== 2024-2025 ===")
print(df_2024["2.r_modelo_de_trabalho_atual"].value_counts(dropna=False))

print("\n=== 2025-2026 ===")
print(df_2025["2.q_modelo_de_trabalho_atual"].value_counts(dropna=False))

=== 2023-2024 ===
('P2_r ', 'Atualmente qual a sua forma de trabalho?')
Modelo 100% remoto                                                                                                2201
Modelo híbrido flexível (o funcionário tem liberdade para escolher quando estar no escritório presencialmente)     972
Modelo híbrido com dias fixos de trabalho presencial                                                               790
Modelo 100% presencial                                                                                             790
NaN                                                                                                                540
Name: count, dtype: int64

=== 2024-2025 ===
2.r_modelo_de_trabalho_atual
Modelo 100% remoto                                                                                                2222
Modelo híbrido flexível (o funcionário tem liberdade para escolher quando estar no escritório presencialmente)     999
Modelo híbrido com di

### Interpretação da Variável Modelo de Trabalho

A comparação da variável relacionada ao modelo de trabalho demonstra consistência entre as três edições da pesquisa.

Foram identificadas as mesmas quatro categorias principais em todos os períodos: trabalho 100% remoto, trabalho 100% presencial, modelo híbrido com dias fixos de trabalho presencial e modelo híbrido flexível.

Embora a quantidade de respostas varie entre as edições, não foram identificadas diferenças relevantes na codificação das categorias.

Dessa forma, as variáveis `P2_r` (2023-2024), `2.r_modelo_de_trabalho_atual` (2024-2025) e `2.q_modelo_de_trabalho_atual` (2025-2026) podem ser harmonizadas sob o atributo padronizado `modelo_trabalho`.

### 6.2 Validação das Variáveis Demográficas

As variáveis de gênero e raça/etnia são comparadas entre as três pesquisas para verificar a consistência das categorias e identificar eventuais necessidades de padronização antes de sua utilização nas análises de diversidade.

In [28]:
# Localização automática das colunas de 2023
coluna_genero_2023 = [
    coluna for coluna in df_2023.columns
    if "Genero" in str(coluna)
][0]

coluna_raca_2023 = [
    coluna for coluna in df_2023.columns
    if "Cor/raca/etnia" in str(coluna)
][0]


print("========== GÊNERO ==========\n")

print("2023-2024")
print(df_2023[coluna_genero_2023].value_counts(dropna=False))

print("\n2024-2025")
print(df_2024["1.b_genero"].value_counts(dropna=False))

print("\n2025-2026")
print(df_2025["1.b_genero"].value_counts(dropna=False))


print("\n\n========== RAÇA/ETNIA ==========\n")

print("2023-2024")
print(df_2023[coluna_raca_2023].value_counts(dropna=False))

print("\n2024-2025")
print(df_2024["1.c_cor/raca/etnia"].value_counts(dropna=False))

print("\n2025-2026")
print(df_2025["1.c_cor/raca/etnia"].value_counts(dropna=False))

========== GÊNERO ==========

2023-2024
('P1_b ', 'Genero')
Masculino               3975
Feminino                1293
Prefiro não informar      16
Outro                      9
Name: count, dtype: int64

2024-2025
1.b_genero
Masculino               3968
Feminino                1226
Prefiro não informar      15
Outro                      8
Name: count, dtype: int64

2025-2026
1.b_genero
Masculino               2707
Feminino                 768
Prefiro não informar      13
Outro                      7
Name: count, dtype: int64


========== RAÇA/ETNIA ==========

2023-2024
('P1_c ', 'Cor/raca/etnia')
Branca                  3414
Parda                   1281
Preta                    387
Amarela                  147
Prefiro não informar      34
Outra                     18
Indígena                  12
Name: count, dtype: int64

2024-2025
1.c_cor/raca/etnia
Branca                  3478
Parda                   1180
Preta                    352
Amarela                  155
Prefiro não informar 

### Interpretação das Variáveis Demográficas

A comparação das variáveis de gênero e raça/etnia demonstra elevada consistência entre as três edições da pesquisa.

Para gênero, foram identificadas as mesmas quatro categorias em todos os períodos: `Masculino`, `Feminino`, `Prefiro não informar` e `Outro`.

Da mesma forma, a variável de raça/etnia mantém as categorias `Branca`, `Parda`, `Preta`, `Amarela`, `Indígena`, `Outra` e `Prefiro não informar` nas três pesquisas. Na edição de 2025-2026 foram identificados apenas dois registros sem preenchimento nessa variável.

Dessa forma, as variáveis apresentam compatibilidade semântica e categórica suficiente para serem harmonizadas sob os atributos `genero` e `raca_etnia` na camada Silver.

Essa padronização permitirá a construção de análises comparativas relacionadas à diversidade no mercado brasileiro de Dados.

### 6.3 Identificação das Variáveis de Perfil Profissional

Para caracterizar a estrutura do mercado e os perfis profissionais presentes nas pesquisas, são identificadas variáveis relacionadas a cargo, senioridade e experiência profissional.

Como a nomenclatura e a organização do questionário variam entre as edições, inicialmente são localizadas as possíveis variáveis relacionadas a esses conceitos antes da definição das correspondências que integrarão o schema harmonizado.

In [29]:
termos_perfil = [
    "cargo",
    "senior",
    "experiencia",
    "experiência",
    "nivel",
    "nível"
]

perfil_2023 = localizar_colunas(df_2023, termos_perfil)
perfil_2024 = localizar_colunas(df_2024, termos_perfil)
perfil_2025 = localizar_colunas(df_2025, termos_perfil)

print("=== 2023-2024 ===")
for coluna in perfil_2023:
    print(coluna)

print("\n=== 2024-2025 ===")
for coluna in perfil_2024:
    print(coluna)

print("\n=== 2025-2026 ===")
for coluna in perfil_2025:
    print(coluna)

=== 2023-2024 ===
('P1_e ', 'experiencia_profissional_prejudicada')
('P1_e_1 ', 'Não acredito que minha experiência profissional seja afetada')
('P1_e_2 ', 'Experiencia prejudicada devido a minha Cor Raça Etnia')
('P1_e_3 ', 'Experiencia prejudicada devido a minha identidade de gênero')
('P1_e_4 ', 'Experiencia prejudicada devido ao fato de ser PCD')
('P1_f_2', 'Senioridade das vagas recebidas em relação à sua experiência')
('P1_f_6', 'Nível de cobrança no trabalho/Stress no trabalho')
('P1_l ', 'Nivel de Ensino')
('P2_e ', 'Cargo como Gestor')
('P2_f ', 'Cargo Atual')
('P2_g ', 'Nivel')
('P2_i ', 'Quanto tempo de experiência na área de dados você tem?')
('P2_j ', 'Quanto tempo de experiência na área de TI/Engenharia de Software você teve antes de começar a trabalhar na área de dados?')
('P3_b ', 'Quais desses papéis/cargos fazem parte do time (ou chapter) de dados da sua empresa?')
('P4_a ', 'Mesmo que esse não seja seu cargo formal, você considera que sua atuação no dia a dia, reflet

### Interpretação das Variáveis de Perfil Profissional

A identificação das variáveis relacionadas ao perfil profissional revelou elevada continuidade semântica entre as três edições da pesquisa.

Foram encontradas correspondências diretas para três dimensões fundamentais do projeto: cargo atual, nível profissional e tempo de experiência na área de Dados. Embora a nomenclatura técnica das colunas tenha sido alterada entre 2023-2024 e as edições posteriores, o significado das variáveis foi preservado.

Esses atributos são particularmente relevantes para a caracterização da estrutura do mercado brasileiro de Dados e para a identificação dos perfis profissionais presentes nas pesquisas.

Antes de sua incorporação definitiva ao schema harmonizado, serão avaliadas as categorias existentes em cada edição para identificar eventuais diferenças de classificação que necessitem de padronização na camada Silver.

### 6.4 Validação das Variáveis de Perfil Profissional

As categorias das variáveis de cargo atual, nível profissional e tempo de experiência em Dados são comparadas entre as três edições para confirmar sua compatibilidade e identificar eventuais alterações no questionário.

In [30]:
# Localização automática das colunas de 2023-2024

coluna_cargo_2023 = [
    coluna for coluna in df_2023.columns
    if "Cargo Atual" in str(coluna)
][0]

coluna_nivel_2023 = [
    coluna for coluna in df_2023.columns
    if "Nivel" in str(coluna) and "P2_g" in str(coluna)
][0]

coluna_experiencia_2023 = [
    coluna for coluna in df_2023.columns
    if "Quanto tempo de experiência na área de dados" in str(coluna)
][0]


# CARGO
print("========== CARGO ATUAL ==========\n")

print("2023-2024")
print(df_2023[coluna_cargo_2023].value_counts(dropna=False))

print("\n2024-2025")
print(df_2024["2.f_cargo_atual"].value_counts(dropna=False))

print("\n2025-2026")
print(df_2025["2.f_cargo_atual"].value_counts(dropna=False))


# SENIORIDADE
print("\n\n========== SENIORIDADE ==========\n")

print("2023-2024")
print(df_2023[coluna_nivel_2023].value_counts(dropna=False))

print("\n2024-2025")
print(df_2024["2.g_nivel"].value_counts(dropna=False))

print("\n2025-2026")
print(df_2025["2.g_nivel"].value_counts(dropna=False))


# EXPERIÊNCIA EM DADOS
print("\n\n========== EXPERIÊNCIA EM DADOS ==========\n")

print("2023-2024")
print(df_2023[coluna_experiencia_2023].value_counts(dropna=False))

print("\n2024-2025")
print(df_2024["2.i_tempo_de_experiencia_em_dados"].value_counts(dropna=False))

print("\n2025-2026")
print(df_2025["2.i_tempo_de_experiencia_em_dados"].value_counts(dropna=False))

========== CARGO ATUAL ==========

2023-2024
('P2_f ', 'Cargo Atual')
NaN                                                                    1436
Analista de Dados/Data Analyst                                          907
Cientista de Dados/Data Scientist                                       687
Engenheiro de Dados/Arquiteto de Dados/Data Engineer/Data Architect     684
Analista de BI/BI Analyst                                               506
Outra Opção                                                             262
Analista de Negócios/Business Analyst                                   195
Analytics Engineer                                                      139
Desenvolvedor/ Engenheiro de Software/ Analista de Sistemas             105
Data Product Manager/ Product Manager (PM/APM/DPM/GPM/PO)                81
Engenheiro de Machine Learning/ML Engineer/AI Engineer                   75
Analista de Suporte/Analista Técnico                                     72
Analista de Inteli

### Interpretação das Variáveis de Perfil Profissional

As três dimensões analisadas apresentam correspondência semântica entre as edições, porém foram identificadas alterações relevantes nas categorias que deverão ser consideradas durante a harmonização dos dados.

Na variável de cargo atual, as principais funções permanecem comparáveis entre os períodos. Entretanto, em 2023-2024 a categoria de Engenharia e Arquitetura de Dados aparece agrupada, enquanto nas pesquisas posteriores também é disponibilizada uma categoria específica para Arquitetura de Dados. Essa diferença deverá ser considerada nas análises comparativas.

A variável de senioridade apresenta as categorias Júnior, Pleno e Sênior nas três pesquisas. Na edição de 2025-2026 foi adicionada a categoria `Especialista/Staff+`, indicando uma ampliação da classificação utilizada pelo questionário. Essa categoria será preservada para evitar perda de informação.

Para experiência na área de Dados, as categorias apresentam elevada continuidade entre as edições. Entretanto, a pesquisa de 2023-2024 apresenta simultaneamente as faixas `de 4 a 6 anos` e `de 5 a 6 anos`, enquanto as pesquisas posteriores utilizam apenas `de 5 a 6 anos`. Essa diferença será considerada na definição das regras de padronização da camada Silver.

Dessa forma, `cargo_atual`, `senioridade` e `experiencia_dados` são consideradas dimensões adequadas para compor o schema harmonizado, com regras específicas de tratamento documentadas para as diferenças identificadas.

In [31]:
novas_variaveis = pd.DataFrame({
    "Variável Padronizada": [
        "cargo_atual",
        "senioridade",
        "experiencia_dados"
    ],

    "2023-2024": [
        "P2_f - Cargo Atual",
        "P2_g - Nivel",
        "P2_i - Tempo de experiência na área de dados"
    ],

    "2024-2025": [
        "2.f_cargo_atual",
        "2.g_nivel",
        "2.i_tempo_de_experiencia_em_dados"
    ],

    "2025-2026": [
        "2.f_cargo_atual",
        "2.g_nivel",
        "2.i_tempo_de_experiencia_em_dados"
    ]
})

schema_harmonizado = pd.concat(
    [schema_harmonizado, novas_variaveis],
    ignore_index=True
)

schema_harmonizado

,Variável Padronizada,2023-2024,2024-2025,2025-2026
0,genero,P1_b - Genero,1.b_genero,1.b_genero
1,raca_etnia,P1_c - Cor/raca/etnia,1.c_cor/raca/etnia,1.c_cor/raca/etnia
2,faixa_salarial,P2_h - Faixa salarial,2.h_faixa_salarial,2.h_faixa_salarial
3,modelo_trabalho,P2_r - Atualmente qual a sua forma de trabalho?,2.r_modelo_de_trabalho_atual,2.q_modelo_de_trabalho_atual
4,cargo_atual,P2_f - Cargo Atual,2.f_cargo_atual,2.f_cargo_atual
5,senioridade,P2_g - Nivel,2.g_nivel,2.g_nivel
6,experiencia_dados,P2_i - Tempo de experiência na área de dados,2.i_tempo_de_experiencia_em_dados,2.i_tempo_de_experiencia_em_dados


### 6.5 Identificação das Variáveis Geográficas

Para permitir análises das diferenças regionais no mercado brasileiro de Dados, são identificadas as variáveis relacionadas à localização dos respondentes.

O objetivo é determinar quais atributos geográficos apresentam correspondência entre as três edições e podem ser utilizados na construção do schema harmonizado.

In [32]:
termos_localizacao = [
    "estado",
    "regiao",
    "região",
    "uf",
    "onde_mora",
    "onde mora"
]

localizacao_2023 = localizar_colunas(df_2023, termos_localizacao)
localizacao_2024 = localizar_colunas(df_2024, termos_localizacao)
localizacao_2025 = localizar_colunas(df_2025, termos_localizacao)

print("=== 2023-2024 ===")
for coluna in localizacao_2023:
    print(coluna)

print("\n=== 2024-2025 ===")
for coluna in localizacao_2024:
    print(coluna)

print("\n=== 2025-2026 ===")
for coluna in localizacao_2025:
    print(coluna)

=== 2023-2024 ===
('P1_i ', 'Estado onde mora')
('P1_i_1 ', 'uf onde mora')
('P1_i_2 ', 'Regiao onde mora')
('P1_j ', 'Mudou de Estado?')
('P1_k ', 'Regiao de origem')

=== 2024-2025 ===
1.i.1_uf_onde_mora
1.i.2_regiao_onde_mora
1.k.1_uf_de_origem
1.k.2_regiao_de_origem
1.h_pais_onde_mora
1.i_estado_onde_mora
1.j_vive_no_estado_de_formacao
1.k_estado_de_origem

=== 2025-2026 ===
1.h_pais_onde_mora
1.i_estado_onde_mora
1.i.1_uf_onde_mora
1.i.2_regiao_onde_mora
1.j_vive_no_estado_de_formacao
1.k_estado_de_origem
1.k.1_uf_de_origem
1.k.2_regiao_de_origem


### Interpretação das Variáveis Geográficas

A identificação das variáveis geográficas revelou correspondência direta entre as três edições da pesquisa para a Unidade Federativa (UF) e a região de residência dos respondentes.

Embora os questionários também disponibilizem informações relacionadas ao local de origem e ao país de residência, foram priorizadas as variáveis referentes à localização atual do profissional, por apresentarem maior aderência às análises regionais propostas no Tech Challenge.

Dessa forma, as variáveis relacionadas à UF e à região onde o respondente mora são candidatas à harmonização sob os atributos padronizados `uf` e `regiao`.

Essas dimensões permitirão comparar indicadores profissionais, remuneração, senioridade, modelos de trabalho e adoção tecnológica entre diferentes regiões brasileiras.

### 6.6 Validação das Variáveis Geográficas

As categorias de região são comparadas entre as três pesquisas para verificar sua consistência antes da incorporação ao schema harmonizado.

In [33]:
# Localização automática da região em 2023-2024
coluna_regiao_2023 = [
    coluna for coluna in df_2023.columns
    if "Regiao onde mora" in str(coluna)
][0]

print("========== REGIÃO ==========\n")

print("2023-2024")
print(df_2023[coluna_regiao_2023].value_counts(dropna=False))

print("\n2024-2025")
print(df_2024["1.i.2_regiao_onde_mora"].value_counts(dropna=False))

print("\n2025-2026")
print(df_2025["1.i.2_regiao_onde_mora"].value_counts(dropna=False))

========== REGIÃO ==========

2023-2024
('P1_i_2 ', 'Regiao onde mora')
Sudeste         3173
Sul              961
Nordeste         608
Centro-oeste     344
NaN              124
Norte             83
Name: count, dtype: int64

2024-2025
1.i.2_regiao_onde_mora
Sudeste         3132
Sul             1031
Nordeste         516
Centro-oeste     332
NaN              142
Norte             64
Name: count, dtype: int64

2025-2026
1.i.2_regiao_onde_mora
Sudeste         2171
Sul              540
Nordeste         380
Centro-oeste     231
NaN              125
Norte             48
Name: count, dtype: int64


### Interpretação das Variáveis Geográficas

A comparação da variável de região demonstra consistência categórica entre as três edições da pesquisa.

As cinco regiões brasileiras — Sudeste, Sul, Nordeste, Centro-oeste e Norte — são representadas com a mesma nomenclatura nos três períodos analisados. Foram identificados alguns registros sem informação geográfica, que serão tratados conforme as regras definidas posteriormente para a camada Silver.

Dessa forma, a dimensão `regiao` apresenta compatibilidade suficiente para utilização direta no schema harmonizado. A variável de UF também será preservada como `uf`, permitindo análises geográficas em diferentes níveis de granularidade.

Essas dimensões possibilitarão avaliar diferenças regionais relacionadas a perfil profissional, remuneração, senioridade, modelo de trabalho e adoção tecnológica.

In [34]:
novas_variaveis_geo = pd.DataFrame({
    "Variável Padronizada": [
        "uf",
        "regiao"
    ],

    "2023-2024": [
        "P1_i_1 - UF onde mora",
        "P1_i_2 - Regiao onde mora"
    ],

    "2024-2025": [
        "1.i.1_uf_onde_mora",
        "1.i.2_regiao_onde_mora"
    ],

    "2025-2026": [
        "1.i.1_uf_onde_mora",
        "1.i.2_regiao_onde_mora"
    ]
})

schema_harmonizado = pd.concat(
    [schema_harmonizado, novas_variaveis_geo],
    ignore_index=True
)

schema_harmonizado

,Variável Padronizada,2023-2024,2024-2025,2025-2026
0,genero,P1_b - Genero,1.b_genero,1.b_genero
1,raca_etnia,P1_c - Cor/raca/etnia,1.c_cor/raca/etnia,1.c_cor/raca/etnia
2,faixa_salarial,P2_h - Faixa salarial,2.h_faixa_salarial,2.h_faixa_salarial
3,modelo_trabalho,P2_r - Atualmente qual a sua forma de trabalho?,2.r_modelo_de_trabalho_atual,2.q_modelo_de_trabalho_atual
4,cargo_atual,P2_f - Cargo Atual,2.f_cargo_atual,2.f_cargo_atual
5,senioridade,P2_g - Nivel,2.g_nivel,2.g_nivel
6,experiencia_dados,P2_i - Tempo de experiência na área de dados,2.i_tempo_de_experiencia_em_dados,2.i_tempo_de_experiencia_em_dados
7,uf,P1_i_1 - UF onde mora,1.i.1_uf_onde_mora,1.i.1_uf_onde_mora
8,regiao,P1_i_2 - Regiao onde mora,1.i.2_regiao_onde_mora,1.i.2_regiao_onde_mora


## 7. Mapeamento das Variáveis de Tecnologia e Inteligência Artificial

Além das dimensões demográficas e profissionais, o Tech Challenge requer a análise das tecnologias adotadas pelos profissionais e da utilização de Inteligência Artificial.

Diferentemente das dimensões anteriores, essas informações podem estar distribuídas em múltiplas variáveis do questionário. Dessa forma, inicialmente são identificados os atributos relacionados às principais tecnologias e ferramentas utilizadas pelos respondentes.

### 7.1 Identificação das Variáveis de Tecnologia

In [35]:
termos_tecnologia = [
    "tecnologia",
    "tecnologias",
    "linguagem",
    "linguagens",
    "banco de dados",
    "cloud",
    "bi",
    "ferramenta"
]

tecnologia_2023 = localizar_colunas(df_2023, termos_tecnologia)
tecnologia_2024 = localizar_colunas(df_2024, termos_tecnologia)
tecnologia_2025 = localizar_colunas(df_2025, termos_tecnologia)

print(f"2023-2024: {len(tecnologia_2023)} variáveis encontradas")
print(f"2024-2025: {len(tecnologia_2024)} variáveis encontradas")
print(f"2025-2026: {len(tecnologia_2025)} variáveis encontradas")

2023-2024: 68 variáveis encontradas
2024-2025: 68 variáveis encontradas
2025-2026: 66 variáveis encontradas


In [36]:
print("=== 2023-2024 ===")
for coluna in tecnologia_2023:
    print(coluna)

print("\n=== 2024-2025 ===")
for coluna in tecnologia_2024:
    print(coluna)

print("\n=== 2025-2026 ===")
for coluna in tecnologia_2025:
    print(coluna)

=== 2023-2024 ===
('P1_f_1', 'Quantidade de oportunidades de emprego/vagas recebidas')
('P1_f_2', 'Senioridade das vagas recebidas em relação à sua experiência')
('P2_l_6 ', 'O clima de trabalho/ambiente não é bom')
('P2_o_4 ', 'Flexibilidade de trabalho remoto')
('P2_o_5 ', 'Ambiente e clima de trabalho')
('P2_o_8 ', 'Maturidade da empresa em termos de tecnologia e dados')
('P3_b_6 ', 'Analista de Business Intelligence/BI')
('P3_c ', 'Quais dessas responsabilidades fazem parte da sua rotina atual de trabalho como gestor?')
('P3_c_4 ', 'Decisão sobre contratação de ferramentas e tecnologias relacionadas a dados.')
('P3_c_5 ', 'Sou gestor da equipe responsável pela engenharia de dados e por manter o Data Lake da empresa como fonte única dos dados, garantindo a qualidade e confiabilidade da informação.')
('P3_d_4 ', 'd Gestão de equipes no ambiente remoto.')
('P3_d_6 ', 'f Organizar as informações e garantir a qualidade e confiabilidade.')
('P3_g_2 ', 'Falta de confiabilidade das saídas 

### Interpretação das Variáveis de Tecnologia

A inspeção das variáveis relacionadas a tecnologias mostra que esse tema está distribuído em diferentes blocos do questionário, contemplando linguagens de programação, serviços de Cloud, ferramentas de BI, tecnologias de ETL, plataformas de Data Lake e Data Warehouse, além de ferramentas utilizadas por profissionais de Data Science.

Também foram observadas alterações na codificação das perguntas entre as edições. Por exemplo, variáveis relacionadas a Cloud e ferramentas de BI mantêm significado semelhante, mas apresentam identificadores diferentes em 2025-2026 em comparação com as pesquisas anteriores.

Dessa forma, a análise tecnológica não será representada por uma única variável consolidada. O schema analítico deverá preservar diferentes dimensões tecnológicas, permitindo avaliar separadamente a adoção de linguagens, Cloud, BI e outras categorias relevantes.

A harmonização será orientada pelas perguntas de negócio do Tech Challenge e pela comparabilidade entre as três edições.

### 7.2 Validação das Variáveis de Linguagens de Programação

As variáveis relacionadas às linguagens de programação são analisadas entre as três edições para verificar como o questionário registra a utilização e a preferência dos profissionais.

In [37]:
print("=== 2023-2024 ===")
for coluna in df_2023.columns:
    if "linguagem" in str(coluna).lower():
        print(coluna)

print("\n=== 2024-2025 ===")
for coluna in df_2024.columns:
    if "linguagem" in str(coluna).lower():
        print(coluna)

print("\n=== 2025-2026 ===")
for coluna in df_2025.columns:
    if "linguagem" in str(coluna).lower():
        print(coluna)

=== 2023-2024 ===
('P4_d_15 ', 'Não utilizo nenhuma linguagem')
('P6_a_3 ', 'Crio consultas através da linguagem SQL para exportar informações e compartilhar com as áreas de negócio.')
('P6_h_3 ', 'Criando consultas através da linguagem SQL para exportar informações e compartilhar com as áreas de negócio.')
('P7_a_3 ', 'Crio consultas através da linguagem SQL para exportar informações e compartilhar com as áreas de negócio.')
('P7_d_3 ', 'Criando consultas através da linguagem SQL para exportar informações e compartilhar com as áreas de negócio.')

=== 2024-2025 ===
4.d_linguagem_de_programacao_(dia_a_dia)
4.e_linguagem_mais_usada
4.f_linguagem_preferida
6.a.3_Crio consultas através da linguagem SQL para exportar informações e compartilhar com as áreas de negócio.
6.h.3_Criando consultas através da linguagem SQL para exportar informações e compartilhar com as áreas de negócio.
7.a.3_Crio consultas através da linguagem SQL para exportar informações e compartilhar com as áreas de negócio

### 7.3 Inspeção do Bloco Tecnológico

A busca textual revelou diferenças na nomenclatura das variáveis de linguagens entre as edições. Para evitar conclusões baseadas apenas nos nomes das colunas, é realizada uma inspeção do bloco tecnológico do questionário, buscando identificar a estrutura das perguntas e possíveis variáveis equivalentes entre os períodos.

In [38]:
print("=== BLOCO 4 | 2023-2024 ===")
for coluna in df_2023.columns:
    if "P4_" in str(coluna):
        print(coluna)

print("\n=== BLOCO 4 | 2024-2025 ===")
for coluna in df_2024.columns:
    if str(coluna).startswith("4."):
        print(coluna)

print("\n=== BLOCO 4 | 2025-2026 ===")
for coluna in df_2025.columns:
    if str(coluna).startswith("4."):
        print(coluna)

=== BLOCO 4 | 2023-2024 ===
('P4_a ', 'Mesmo que esse não seja seu cargo formal, você considera que sua atuação no dia a dia, reflete alguma das opções listadas abaixo?')
('P4_a_1 ', 'Atuacao')
('P4_b ', 'Quais das fontes de dados listadas você já analisou ou processou no trabalho?')
('P4_b_1 ', 'Dados relacionais (estruturados em bancos SQL)')
('P4_b_2 ', 'Dados armazenados em bancos NoSQL')
('P4_b_3 ', 'Imagens')
('P4_b_4 ', 'Textos/Documentos')
('P4_b_5 ', 'Vídeos')
('P4_b_6 ', 'Áudios')
('P4_b_7 ', 'Planilhas')
('P4_b_8 ', 'Dados georeferenciados')
('P4_c ', 'Entre as fontes de dados listadas, quais você utiliza na maior parte do tempo?')
('P4_c_1 ', 'Dados relacionais (estruturados em bancos SQL)')
('P4_c_2 ', 'Dados armazenados em bancos NoSQL')
('P4_c_3 ', 'Imagens')
('P4_c_4 ', 'Textos/Documentos')
('P4_c_5 ', 'Vídeos')
('P4_c_6 ', 'Áudios')
('P4_c_7 ', 'Planilhas')
('P4_c_8 ', 'Dados georeferenciados')
('P4_d ', 'Quais das linguagens listadas abaixo você utiliza no trabalho?')

### Interpretação da Estrutura do Bloco Tecnológico

A inspeção detalhada do bloco tecnológico demonstrou elevada continuidade entre as três edições da pesquisa, embora tenham ocorrido alterações na codificação e no posicionamento das perguntas.

Foram identificados grupos comparáveis relacionados a bancos de dados, serviços de Cloud, ferramentas de Business Intelligence e utilização de Inteligência Artificial Generativa. Nessas dimensões, apesar da mudança dos identificadores das colunas, as categorias apresentam correspondência semântica suficiente para harmonização.

A dimensão de linguagens de programação requer tratamento específico. Nas edições 2023-2024 e 2024-2025, o questionário registra explicitamente as linguagens utilizadas no trabalho, enquanto em 2025-2026 o bloco disponível está associado à linguagem preferida. Dessa forma, não será assumida equivalência direta entre uso e preferência na construção das análises temporais.

Essa distinção será preservada para evitar comparações conceitualmente inconsistentes entre os períodos.

### 7.4 Validação das Tecnologias de Cloud

A utilização de serviços de Cloud constitui uma dimensão comparável entre as três pesquisas. Nesta etapa, são verificadas as categorias disponíveis e suas respectivas frequências, permitindo avaliar posteriormente a evolução da adoção das principais plataformas de nuvem.

In [39]:
cloud_2023 = [
    coluna for coluna in df_2023.columns
    if str(coluna).startswith("('P4_h_")
]

cloud_2024 = [
    coluna for coluna in df_2024.columns
    if str(coluna).startswith("4.h.")
]

cloud_2025 = [
    coluna for coluna in df_2025.columns
    if str(coluna).startswith("4.e.")
]

print("Quantidade de categorias:")
print("2023-2024:", len(cloud_2023))
print("2024-2025:", len(cloud_2024))
print("2025-2026:", len(cloud_2025))

print("\n2023-2024:")
for c in cloud_2023:
    print(c)

print("\n2024-2025:")
for c in cloud_2024:
    print(c)

print("\n2025-2026:")
for c in cloud_2025:
    print(c)

Quantidade de categorias:
2023-2024: 7
2024-2025: 7
2025-2026: 7

2023-2024:
('P4_h_1 ', 'Azure (Microsoft)')
('P4_h_2 ', 'Amazon Web Services (AWS)')
('P4_h_3 ', 'Google Cloud (GCP)')
('P4_h_4 ', 'Oracle Cloud')
('P4_h_5 ', 'IBM')
('P4_h_6 ', 'Servidores On Premise/Não utilizamos Cloud')
('P4_h_7 ', 'Cloud Própria')

2024-2025:
4.h.1_Amazon Web Services (AWS)
4.h.2_Google Cloud (GCP)
4.h.3_Azure (Microsoft)
4.h.4_Oracle Cloud
4.h.5_IBM
4.h.6_Servidores On Premise/Não utilizamos Cloud
4.h.7_Cloud Própria

2025-2026:
4.e.1_Amazon Web Services (AWS)
4.e.2_Google Cloud (GCP)
4.e.3_Azure (Microsoft)
4.e.4_Oracle Cloud
4.e.5_IBM
4.e.6_Servidores On Premise/Não utilizamos Cloud
4.e.7_Cloud Própria


### Interpretação das Variáveis de Cloud

A análise das variáveis relacionadas à infraestrutura de Cloud revelou elevada consistência entre as três edições da pesquisa.

As mesmas sete categorias estão presentes nos três períodos: Amazon Web Services (AWS), Microsoft Azure, Google Cloud (GCP), Oracle Cloud, IBM, servidores On Premise/ausência de Cloud e Cloud própria.

Embora a ordem e os identificadores das colunas tenham sido alterados entre as edições, o significado das categorias foi preservado. Dessa forma, a harmonização deverá ser realizada com base no significado das categorias, e não na posição das colunas no dataset.

Essa consistência permite utilizar a dimensão de Cloud em análises temporais de adoção tecnológica entre as três pesquisas.

### 7.5 Estrutura dos Indicadores de Cloud

Após confirmar a correspondência das categorias entre as pesquisas, são inspecionados os valores armazenados nas variáveis de Cloud. O objetivo é compreender a codificação utilizada para representar a seleção de cada tecnologia antes da definição das regras de transformação.

In [40]:
print("=== 2023-2024 ===")
for coluna in cloud_2023:
    print("\n", coluna)
    print(df_2023[coluna].value_counts(dropna=False).head())

print("\n=== 2024-2025 ===")
for coluna in cloud_2024:
    print("\n", coluna)
    print(df_2024[coluna].value_counts(dropna=False).head())

print("\n=== 2025-2026 ===")
for coluna in cloud_2025:
    print("\n", coluna)
    print(df_2025[coluna].value_counts(dropna=False).head())

=== 2023-2024 ===

 ('P4_h_1 ', 'Azure (Microsoft)')
('P4_h_1 ', 'Azure (Microsoft)')
0.0    2084
NaN    1680
1.0    1529
Name: count, dtype: int64

 ('P4_h_2 ', 'Amazon Web Services (AWS)')
('P4_h_2 ', 'Amazon Web Services (AWS)')
0.0    2507
NaN    1680
1.0    1106
Name: count, dtype: int64

 ('P4_h_3 ', 'Google Cloud (GCP)')
('P4_h_3 ', 'Google Cloud (GCP)')
0.0    2472
NaN    1680
1.0    1141
Name: count, dtype: int64

 ('P4_h_4 ', 'Oracle Cloud')
('P4_h_4 ', 'Oracle Cloud')
0.0    3459
NaN    1680
1.0     154
Name: count, dtype: int64

 ('P4_h_5 ', 'IBM')
('P4_h_5 ', 'IBM')
0.0    3566
NaN    1680
1.0      47
Name: count, dtype: int64

 ('P4_h_6 ', 'Servidores On Premise/Não utilizamos Cloud')
('P4_h_6 ', 'Servidores On Premise/Não utilizamos Cloud')
0.0    3031
NaN    1680
1.0     582
Name: count, dtype: int64

 ('P4_h_7 ', 'Cloud Própria')
('P4_h_7 ', 'Cloud Própria')
0.0    3362
NaN    1680
1.0     251
Name: count, dtype: int64

=== 2024-2025 ===

 4.h.1_Amazon Web Services (AW

### Interpretação da Codificação das Tecnologias de Cloud

As variáveis relacionadas às tecnologias de Cloud utilizam codificação binária, na qual o valor `1` representa a seleção da tecnologia e o valor `0` representa sua não seleção.

Também foi identificada uma quantidade relevante de valores ausentes (`NaN`). Como a quantidade de valores ausentes é constante entre as diferentes tecnologias dentro de cada edição, esses registros não serão interpretados automaticamente como ausência de utilização da tecnologia. Eles serão preservados como valores ausentes durante a harmonização, evitando a criação artificial de respostas negativas.

A estrutura observada é consistente entre as três pesquisas e permite a padronização das tecnologias de Cloud na camada Silver.

### 7.6 Validação das Ferramentas de Business Intelligence

Após a validação das tecnologias de Cloud, são analisadas as ferramentas de Business Intelligence presentes nas três pesquisas, buscando verificar a continuidade das categorias e sua adequação para análises comparativas de adoção tecnológica.

In [41]:
# Localização das variáveis de BI

bi_2023 = [
    coluna for coluna in df_2023.columns
    if str(coluna).startswith("('P4_j_")
]

bi_2024 = [
    coluna for coluna in df_2024.columns
    if str(coluna).startswith("4.j.")
]

bi_2025 = [
    coluna for coluna in df_2025.columns
    if str(coluna).startswith("4.g.")
]

print("Quantidade de categorias:")
print("2023-2024:", len(bi_2023))
print("2024-2025:", len(bi_2024))
print("2025-2026:", len(bi_2025))

print("\n=== 2023-2024 ===")
for coluna in bi_2023:
    print(coluna)

print("\n=== 2024-2025 ===")
for coluna in bi_2024:
    print(coluna)

print("\n=== 2025-2026 ===")
for coluna in bi_2025:
    print(coluna)

Quantidade de categorias:
2023-2024: 23
2024-2025: 18
2025-2026: 18

=== 2023-2024 ===
('P4_j_1 ', 'Microsoft PowerBI')
('P4_j_2 ', 'Qlik View/Qlik Sense')
('P4_j_3 ', 'Tableau')
('P4_j_4 ', 'Metabase')
('P4_j_5 ', 'Superset')
('P4_j_6 ', 'Redash')
('P4_j_7 ', 'Looker')
('P4_j_8 ', 'Looker Studio(Google Data Studio)')
('P4_j_9 ', 'Amazon Quicksight')
('P4_j_10 ', 'Mode')
('P4_j_11 ', 'Alteryx')
('P4_j_12 ', 'MicroStrategy')
('P4_j_13 ', 'IBM Analytics/Cognos')
('P4_j_14 ', 'SAP Business Objects/SAP Analytics')
('P4_j_15 ', 'Oracle Business Intelligence')
('P4_j_16 ', 'Salesforce/Einstein Analytics')
('P4_j_17 ', 'Birst')
('P4_j_18 ', 'SAS Visual Analytics')
('P4_j_19 ', 'Grafana')
('P4_j_20 ', 'TIBCO Spotfire')
('P4_j_21 ', 'Pentaho')
('P4_j_22 ', 'Fazemos todas as análises utilizando apenas Excel ou planilhas do google')
('P4_j_23 ', 'Não utilizo nenhuma ferramenta de BI no trabalho')

=== 2024-2025 ===
4.j.1_Microsoft PowerBI
4.j.2_Qlik View/Qlik Sense
4.j.3_Tableau
4.j.4_Metabase
4.

### Interpretação das Ferramentas de Business Intelligence

A comparação das ferramentas de Business Intelligence identificou uma alteração na quantidade de categorias disponíveis entre as edições da pesquisa.

A edição 2023-2024 apresenta 23 opções, enquanto as pesquisas 2024-2025 e 2025-2026 apresentam 18 categorias. Algumas ferramentas presentes na primeira edição analisada, como Mode, MicroStrategy, IBM Analytics/Cognos, Birst e TIBCO Spotfire, deixaram de aparecer nas edições posteriores.

Apesar dessa alteração, foi identificado um núcleo de 18 categorias com continuidade entre os três períodos, incluindo ferramentas como Microsoft Power BI, Qlik View/Qlik Sense, Tableau, Metabase, Superset, Looker, Looker Studio, Amazon QuickSight, Grafana e Pentaho, além das categorias relacionadas ao uso exclusivo de planilhas e à ausência de ferramentas de BI.

Para análises temporais comparativas, serão priorizadas as categorias presentes nas três edições, evitando interpretar a ausência de determinadas ferramentas nas pesquisas posteriores como ausência de adoção no mercado.

Essa abordagem preserva a comparabilidade histórica e reduz possíveis distorções provocadas por alterações no questionário.

### 7.7 Identificação das Variáveis de Inteligência Artificial

A utilização de Inteligência Artificial constitui uma das dimensões centrais da análise tecnológica. Nesta etapa são identificadas as variáveis relacionadas à adoção de IA Generativa e ferramentas baseadas em Large Language Models (LLMs), permitindo avaliar sua comparabilidade entre as três edições.

In [42]:
termos_ia = [
    "generativa",
    "llm",
    "chatgpt",
    "copilot",
    "inteligência artificial",
    "inteligencia artificial"
]

ia_2023 = localizar_colunas(df_2023, termos_ia)
ia_2024 = localizar_colunas(df_2024, termos_ia)
ia_2025 = localizar_colunas(df_2025, termos_ia)

print(f"2023-2024: {len(ia_2023)} variáveis")
print(f"2024-2025: {len(ia_2024)} variáveis")
print(f"2025-2026: {len(ia_2025)} variáveis")

print("\n=== 2023-2024 ===")
for coluna in ia_2023:
    print(coluna)

print("\n=== 2024-2025 ===")
for coluna in ia_2024:
    print(coluna)

print("\n=== 2025-2026 ===")
for coluna in ia_2025:
    print(coluna)

2023-2024: 32 variáveis
2024-2025: 32 variáveis
2025-2026: 33 variáveis

=== 2023-2024 ===
('P3_c_7 ', 'Sou gestor da equipe responsável por iniciativas e projetos envolvendo Inteligência Artificial e Machine Learning.')
('P3_e ', 'AI Generativa é uma prioridade em sua empresa?')
('P3_f ', 'Tipos de uso de AI Generativa e LLMs na empresa')
('P3_f_1 ', 'Colaboradores usando AI generativa de forma independente e descentralizada')
('P3_f_2 ', 'Direcionamento centralizado do uso de AI generativa')
('P3_f_3 ', 'Desenvolvedores utilizando Copilots')
('P3_f_4 ', 'AI Generativa e LLMs para melhorar produtos externos')
('P3_f_5 ', 'AI Generativa e LLMs para melhorar produtos internos para os colaboradores')
('P3_f_6 ', 'IA Generativa e LLMs como principal frente do negócio')
('P3_f_7 ', 'IA Generativa e LLMs não é prioridade')
('P3_f_8 ', 'Não sei opinar sobre o uso de IA Generativa e LLMs na empresa')
('P3_g ', 'Motivos que levam a empresa a não usar AI Genrativa e LLMs')
('P3_g_5 ', 'Retorno 

### Interpretação das Variáveis de Inteligência Artificial

A inspeção das variáveis relacionadas à Inteligência Artificial demonstrou elevada continuidade entre as três edições da pesquisa.

Foram identificados dois grupos particularmente adequados às análises comparativas: o tipo de utilização de IA Generativa e LLMs nas empresas e a utilização de ferramentas como ChatGPT, LLMs e Copilots no trabalho.

Embora os identificadores das variáveis tenham sido alterados ao longo das edições, as categorias associadas a esses dois blocos permanecem semanticamente compatíveis, permitindo sua harmonização e comparação temporal.

Na edição 2025-2026 foi identificada ainda uma nova variável relacionada aos resultados obtidos pelas empresas com LLMs. Por não possuir correspondência direta nas edições anteriores, essa variável poderá ser utilizada como análise complementar do período mais recente, mas não será considerada uma dimensão histórica comparável.

Essa estrutura permitirá avaliar tanto a evolução da adoção individual de ferramentas de IA quanto mudanças nas formas de utilização de IA Generativa pelas organizações.

### 7.8 Validação do Uso de IA Generativa no Trabalho

Para avaliar a evolução da utilização de ferramentas de Inteligência Artificial Generativa pelos profissionais, são comparadas as categorias relacionadas ao uso de ChatGPT, LLMs e soluções do tipo Copilot nas três edições.

In [43]:
ia_trabalho_2023 = [
    coluna for coluna in df_2023.columns
    if str(coluna).startswith("('P4_m_")
]

ia_trabalho_2024 = [
    coluna for coluna in df_2024.columns
    if str(coluna).startswith("4.m.")
]

ia_trabalho_2025 = [
    coluna for coluna in df_2025.columns
    if str(coluna).startswith("4.j.")
]

print("Quantidade de categorias:")
print("2023-2024:", len(ia_trabalho_2023))
print("2024-2025:", len(ia_trabalho_2024))
print("2025-2026:", len(ia_trabalho_2025))

print("\n=== 2023-2024 ===")
for coluna in ia_trabalho_2023:
    print(coluna)

print("\n=== 2024-2025 ===")
for coluna in ia_trabalho_2024:
    print(coluna)

print("\n=== 2025-2026 ===")
for coluna in ia_trabalho_2025:
    print(coluna)

Quantidade de categorias:
2023-2024: 5
2024-2025: 5
2025-2026: 5

=== 2023-2024 ===
('P4_m_1 ', 'Não uso soluções de AI Generativa com foco em produtividade')
('P4_m_2 ', 'Uso soluções gratuitas de AI Generativa com foco em produtividade')
('P4_m_3 ', 'Uso e pago pelas soluções de AI Generativa com foco em produtividade')
('P4_m_4 ', 'A empresa que trabalho paga pelas soluções de AI Generativa com foco em produtividade')
('P4_m_5 ', 'Uso soluções do tipo Copilot')

=== 2024-2025 ===
4.m.1 Não uso soluções de AI Generativa com foco em produtividade
4.m.2 Uso soluções gratuitas de AI Generativa com foco em produtividade
4.m.3 Uso e pago pelas soluções de AI Generativa com foco em produtividade
4.m.4 A empresa que trabalho paga pelas soluções de AI Generativa com foco em produtividade
4.m.5 Uso soluções do tipo Copilot

=== 2025-2026 ===
4.j.1 Não uso soluções de AI Generativa com foco em produtividade
4.j.2 Uso soluções gratuitas de AI Generativa com foco em produtividade
4.j.3 Uso e pag

### Conclusão da Análise de Inteligência Artificial

As variáveis relacionadas ao uso de Inteligência Artificial Generativa no trabalho apresentam elevada consistência entre as três edições analisadas.

Foram identificadas cinco categorias semanticamente equivalentes em todos os períodos: não utilização de soluções de IA Generativa, utilização de soluções gratuitas, contratação individual de soluções pagas, contratação das soluções pela empresa e utilização de ferramentas do tipo Copilot.

Embora os identificadores técnicos das variáveis tenham sido alterados entre as edições, o significado das categorias permaneceu consistente. Dessa forma, essas variáveis poderão ser harmonizadas na etapa de transformação e utilizadas para análises temporais sobre a evolução da adoção de IA Generativa pelos profissionais de dados.

## 7.9 Identificação das Tecnologias de Banco de Dados

Nesta etapa são identificadas as variáveis relacionadas às tecnologias de bancos de dados utilizadas pelos profissionais, buscando verificar a existência de categorias comparáveis entre as três edições da pesquisa.

In [44]:
termos_banco = [
    "banco de dados",
    "database",
    "postgres",
    "mysql",
    "sql server",
    "oracle",
    "mongodb",
    "dynamodb",
    "cassandra",
    "redis",
    "mariadb",
    "sqlite"
]

banco_2023 = localizar_colunas(df_2023, termos_banco)
banco_2024 = localizar_colunas(df_2024, termos_banco)
banco_2025 = localizar_colunas(df_2025, termos_banco)

print("Quantidade de variáveis encontradas:")
print("2023-2024:", len(banco_2023))
print("2024-2025:", len(banco_2024))
print("2025-2026:", len(banco_2025))

print("\n=== 2023-2024 ===")
for coluna in banco_2023:
    print(coluna)

print("\n=== 2024-2025 ===")
for coluna in banco_2024:
    print(coluna)

print("\n=== 2025-2026 ===")
for coluna in banco_2025:
    print(coluna)

Quantidade de variáveis encontradas:
2023-2024: 17
2024-2025: 17
2025-2026: 17

=== 2023-2024 ===
('P3_b_5 ', 'Database Administrator/DBA')
('P4_g_1 ', 'MySQL')
('P4_g_2 ', 'Oracle')
('P4_g_3 ', 'SQL SERVER')
('P4_g_5 ', 'DynamoDB')
('P4_g_7 ', 'Cassandra')
('P4_g_8 ', 'MongoDB')
('P4_g_9 ', 'MariaDB')
('P4_g_12 ', 'PostgreSQL')
('P4_g_16 ', 'SQLite')
('P4_g_20 ', 'Redis')
('P4_h_4 ', 'Oracle Cloud')
('P4_j_15 ', 'Oracle Business Intelligence')
('P6_b_13 ', 'Oracle Data Integrator')
('P6_b_16 ', 'SQL Server Integration Services (SSIS))
('P7_b_13 ', 'Oracle Data Integrator')
('P7_b_16 ', 'SQL Server Integration Services (SSIS)')

=== 2024-2025 ===
3.b.5_Database Administrator/DBA
4.g.1_MySQL
4.g.2_Oracle
4.g.3_SQL SERVER
4.g.5_DynamoDB
4.g.7_Cassandra
4.g.8_MongoDB
4.g.9_MariaDB
4.g.12_PostgreSQL
4.g.16_SQLite
4.g.20_Redis
4.h.4_Oracle Cloud
4.j.12_Oracle Business Intelligence
6.b.13_Oracle Data Integrator
6.b.16_SQL Server Integration Services (SSIS)
7.b.13_Oracle Data Integrator
7.b.1

### Interpretação das Tecnologias de Banco de Dados

A análise das variáveis relacionadas às tecnologias de banco de dados demonstrou elevada consistência entre as três edições da pesquisa.

Foram identificadas categorias equivalentes para tecnologias como MySQL, Oracle, SQL Server, DynamoDB, Cassandra, MongoDB, MariaDB, PostgreSQL, SQLite e Redis. Embora os identificadores técnicos das colunas tenham sido alterados entre os períodos, as categorias analisadas permaneceram semanticamente compatíveis.

Esse resultado permite a harmonização dessas variáveis durante a etapa de transformação, possibilitando análises temporais sobre a utilização de diferentes tecnologias de armazenamento de dados pelos profissionais pesquisados.

As demais ocorrências encontradas pela busca, como Database Administrator (DBA), Oracle Cloud, Oracle Business Intelligence, Oracle Data Integrator e SQL Server Integration Services (SSIS), pertencem a outras dimensões do questionário e não serão consideradas como tecnologias de banco de dados nesta análise.

## 8. Definição do Schema Analítico Proposto

Após a exploração das três edições da pesquisa, foram identificadas as principais dimensões com potencial de comparação temporal.

A próxima etapa consiste em consolidar essas variáveis em um schema analítico comum, independentemente dos diferentes nomes utilizados nos arquivos originais.

Esse mapeamento servirá como referência para a etapa de transformação do pipeline, na qual os dados brutos serão padronizados e convertidos em uma estrutura adequada para análises posteriores.

### 8.1 Estrutura do Schema Harmonizado

O schema analítico proposto consolida as principais dimensões necessárias para responder às questões de negócio do Tech Challenge.

A estrutura prioriza atributos comparáveis entre as três edições da pesquisa e evita a harmonização indiscriminada de todas as centenas de variáveis disponíveis.

As dimensões selecionadas serão utilizadas posteriormente na camada Silver, enquanto estruturas multivaloradas, como tecnologias, ferramentas de BI, Cloud e Inteligência Artificial, poderão ser tratadas em tabelas analíticas específicas.

In [45]:
schema_analitico = pd.DataFrame({
    "Dimensão": [
        "Demografia",
        "Demografia",
        "Carreira",
        "Carreira",
        "Carreira",
        "Remuneração",
        "Modelo de trabalho",
        "Localização",
        "Localização",
        "Tecnologia",
        "Tecnologia",
        "Tecnologia",
        "Tecnologia",
        "Inteligência Artificial",
        "Inteligência Artificial"
    ],

    "Variável Padronizada": [
        "genero",
        "raca_etnia",
        "cargo_atual",
        "senioridade",
        "experiencia_dados",
        "faixa_salarial",
        "modelo_trabalho",
        "uf",
        "regiao",
        "bancos_dados",
        "cloud",
        "ferramentas_bi",
        "linguagem_preferida",
        "uso_ia_generativa",
        "uso_chatgpt_copilot"
    ],

    "Tipo de Estrutura": [
        "Categórica",
        "Categórica",
        "Categórica",
        "Categórica",
        "Categórica ordinal",
        "Categórica ordinal",
        "Categórica",
        "Categórica",
        "Categórica",
        "Multivalorada",
        "Multivalorada",
        "Multivalorada",
        "Categórica",
        "Multivalorada",
        "Multivalorada"
    ],

    "Uso Analítico": [
        "Diversidade",
        "Diversidade",
        "Estrutura do mercado e perfil profissional",
        "Perfil profissional e valorização",
        "Experiência e valorização profissional",
        "Remuneração e valorização",
        "Comparação entre modelos de trabalho",
        "Análises geográficas",
        "Análises regionais",
        "Adoção tecnológica",
        "Adoção tecnológica",
        "Adoção tecnológica",
        "Preferência tecnológica",
        "Adoção de IA nas empresas",
        "Uso individual de IA Generativa"
    ]
})

schema_analitico

,Dimensão,Variável Padronizada,Tipo de Estrutura,Uso Analítico
0,Demografia,genero,Categórica,Diversidade
1,Demografia,raca_etnia,Categórica,Diversidade
2,Carreira,cargo_atual,Categórica,Estrutura do mercado e perfil profissional
3,Carreira,senioridade,Categórica,Perfil profissional e valorização
4,Carreira,experiencia_dados,Categórica ordinal,Experiência e valorização profissional
5,Remuneração,faixa_salarial,Categórica ordinal,Remuneração e valorização
6,Modelo de trabalho,modelo_trabalho,Categórica,Comparação entre modelos de trabalho
7,Localização,uf,Categórica,Análises geográficas
8,Localização,regiao,Categórica,Análises regionais
9,Tecnologia,bancos_dados,Multivalorada,Adoção tecnológica


### Interpretação do Schema Analítico

O schema proposto separa variáveis descritivas do perfil profissional de estruturas tecnológicas multivaloradas.

Dimensões como gênero, raça/etnia, cargo, senioridade, experiência, remuneração, modelo de trabalho e localização poderão compor uma tabela principal harmonizada por respondente.

Já informações como bancos de dados, tecnologias de Cloud, ferramentas de BI e utilização de Inteligência Artificial possuem natureza multivalorada, pois um mesmo respondente pode selecionar diversas opções. Essas dimensões poderão ser modeladas separadamente para facilitar agregações e análises de adoção.

Essa organização busca equilibrar comparabilidade histórica, clareza analítica e flexibilidade para as consultas posteriores.

## 9. Conclusão do Data Understanding

A análise exploratória das pesquisas Estado dos Dados Brasil 2023-2024, 2024-2025 e 2025-2026 permitiu compreender a estrutura dos dados e identificar os principais desafios para sua integração.

As três edições apresentam diferenças na quantidade e na nomenclatura das variáveis, com uma mudança estrutural mais significativa entre 2023-2024 e as edições posteriores. Apesar dessas diferenças, foram identificadas diversas dimensões semanticamente equivalentes, possibilitando a construção de uma visão histórica harmonizada.

Também foram identificados valores ausentes, registros duplicados e alterações pontuais nas categorias disponíveis entre as edições. Esses aspectos foram documentados nesta etapa e serão tratados posteriormente no processo de transformação dos dados.

Com base nas análises realizadas, foi definido um schema analítico contendo dimensões relacionadas a perfil profissional, carreira, remuneração, modelo de trabalho, localização, tecnologias, Cloud, ferramentas de BI, bancos de dados e Inteligência Artificial.

As próximas etapas do projeto serão realizadas por meio de uma arquitetura de dados na AWS, seguindo uma abordagem em camadas:

- **Bronze:** armazenamento dos dados brutos das três pesquisas;
- **Silver:** limpeza, padronização e harmonização dos dados;
- **Gold:** disponibilização de estruturas analíticas preparadas para consultas, indicadores e DataViz.

O presente notebook constitui, portanto, a etapa de Data Understanding e definição das regras que orientarão a construção do pipeline de Big Data & Analytics.